tsconfig.json

```diff
@@ -1,24 +1,31 @@
 {
-  "compileOnSave": false,
-  "compilerOptions": {
-    "allowJs": true,
-    "downlevelIteration": true,
-    "importHelpers": true,
-    "allowSyntheticDefaultImports": true,
-    "outDir": "build",
-    "rootDir": "./", 
-    "sourceMap": true,
-    "strict": true,
-    "noImplicitAny": true,
-    "declaration": false,
-    "moduleResolution": "node",
-    "emitDecoratorMetadata": true,
-    "experimentalDecorators": true,
-    "target": "es2017",
-    "typeRoots": ["node_modules/@types", "api/types"],
-    "lib": ["es6", "dom"],
-    "baseUrl": "./",
-    "module": "es2020",
-    "resolveJsonModule": true  
-  }
+    "compileOnSave": false,
+    "ts-node": {
+        // these options are overrides used only by ts-node
+        // same as our --compilerOptions flag and our TS_NODE_COMPILER_OPTIONS environment variable
+        "compilerOptions": {
+            "module": "commonjs"
+        }
+    },
+    "compilerOptions": {
+        "allowJs": true,
+        "downlevelIteration": true,
+        "importHelpers": true,
+        "allowSyntheticDefaultImports": true,
+        "outDir": "build",
+        "rootDir": "./",
+        "sourceMap": true,
+        "strict": true,
+        "noImplicitAny": true,
+        "declaration": false,
+        "moduleResolution": "node",
+        "emitDecoratorMetadata": true,
+        "experimentalDecorators": true,
+        "target": "es2017",
+        "typeRoots": ["node_modules/@types", "api/types"],
+        "lib": ["es6", "dom"],
+        "baseUrl": "./",
+        "module": "esnext",
```

Comment:
so we don't have a defined module type if that's the case? What if new proposals are introduced, does it automatically translate to the newest version? What aboude node support?